README.md

```diff
@@ -131,6 +131,14 @@ $ cat .gitattributes
 *.rb linguist-language=Java
 ```
 
+Note that, when using Linguist locally, `.gitattributes` must be committed again each time it is modified for Linguist to take it into account. Also note that Linguist requires that you match files and not directories in `.gitattributes`. You might need to use wildcards to match files deeper than top-level ones.
```

Comment:
You changed this before I could point out the pattern matching is identical for `gitattributes` and `gitignore` with the exception that `gitattributes` doesn't support negation 😄.

> Also note that Linguist requires that you match files and not directories...

... doesn't sound correct to me.

From `gitignore`:

```
 o   If the pattern ends with a slash, it is removed for the purpose of the following description, but it would only find a match with a directory. In other
     words, foo/ will match a directory foo and paths underneath it, but will not match a regular file or a symbolic link foo (this is consistent with the
     way how pathspec works in general in Git).

 o   If the pattern does not contain a slash /, Git treats it as a shell glob pattern and checks for a match against the pathname relative to the location
     of the .gitignore file (relative to the toplevel of the work tree if not from a .gitignore file).

 o   Otherwise, Git treats the pattern as a shell glob suitable for consumption by fnmatch(3) with the FNM_PATHNAME flag: wildcards in the pattern will not
     match a / in the pathname. For example, "Documentation/*.html" matches "Documentation/git.html" but not "Documentation/ppc/ppc.html" or
     "tools/perf/Documentation/perf.html".
```

So you should find `dir` will work just as well as `dir/*`, however `dir/` will _not_ because of the trailing `/` which makes git treat it as a directory and not a glob.

Reply:
What if I simply remove the `Linguist requires that you match files and not directories` part then?

Reply:
Hmmm, ignore my comment about `dir` being equivalent to `dir/*`... just tested it and seems it doesn't work, which is news to me. 🤔

Reply:
And really for my use-case, `licenses linguist-vendored` does not work. I found only `**/licenses/* linguist-vendored` to be working :confused: